### PR TITLE
migrate workflows to template handler

### PR DESCRIPTION
**Pull request by davidgamero**
# Description

Migrate the workflow generation to use handler templates

Fixes # (issue)
Feature # (details)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor

## How Has This Been Tested?

Provide instructions so we can reproduce, and list any relevant details for your test configuration. Please mention if this is a breaking change which will impact consuming tool(s).

- [ ] Unit Test:
- [ ] E2E Test:
- [ ] Other Test:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/workflows/workflows.go [hunk 1]
```diff
@@ -1,7 +1,6 @@
 package workflows
```
Comment (bfoley13) [inline pkg/workflows/workflows.go:1]: My plan was to remove these old template handlers completely, check out my PR on the addons migration here: https://github.com/Azure/draft/pull/406/files

I added the custom logic into the cmdhelper package as a placeholder until we have better repo analysis

File: cmd/generate-workflow.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func (gwc *generateWorkflowCmd) generateWorkflows() error {
 	if gwc.deployType == "" {
 		selection := &promptui.Select{
 			Label: "Select k8s Deployment Type",
-			Items: []string{"helm", "kustomize", "manifests"},
+			Items: []string{"helm", "kustomize", "manifest"},
```
Comment (bfoley13) [inline cmd/generate-workflow.go:67]: this is breaking, instead keep it as is and move the azure-pipeline to be "manifests"

https://github.com/Azure/draft/blob/43cdf6b7802fbdb8579ac83c13fb3b074381462e/cmd/info.go#L19C1-L23C2

File: pkg/workflows/workflows.go (hunk 1 above)
Comment (bfoley13) [reply pkg/workflows/workflows.go:1]: Also make sure you remove the `workflow.go` template file

Comment (davidgamero) [reply pkg/workflows/workflows.go]: done

**Review by bfoley13 — APPROVED** [review]

**Review by OliverMKing — APPROVED** [review]

**Pull request merged**